src/libsyntax/ptr.rs

```diff
@@ -95,6 +95,16 @@ impl<T: 'static> P<T> {
     }
 }
 
+impl<T: 'static> P<[T]> {
+    pub fn map_slice<F>(self, f: F) -> P<[T]> where
+        F: FnOnce(Vec<T>) -> Vec<T>
```

Comment:
This looks similar to existing `move_map`/`move_flat_map`.

Reply:
It is similar, but `move_map` and `move_flat_map` expect to receive one element at a time, whereas this takes the whole list at once, allowing it to do things like modifying the last element.

Reply:
Should be able to remove this now.